src/firmware/types.rs

```diff
@@ -202,3 +208,522 @@ pub struct SnpPlatformStatus {
+            reserved: Default::default(),
+        }
+    }
+}
+
+/// The firmware associates each guest with a guest policy that the guest owner provides. The
+/// firmware restricts what actions the hypervisor can take on this guest according to the guest policy.
+/// The policy also indicates the minimum firmware version to for the guest.
+///
+/// The guest owner provides the guest policy to the firmware during launch. The firmware then binds
+/// the policy to the guest. The policy cannot be changed throughout the lifetime of the guest. The
+/// policy is also migrated with the guest and enforced by the destination platform firmware.
+///
+/// | Bit(s) | Name          | Description                                                                                                    |
+/// |--------|---------------|----------------------------------------------------------------------------------------------------------------|
+/// | 7:0    | ABI_MINOR     | The minimum ABI minor version required for this guest to run.                                                  |
+/// | 15:8   | ABI_MAJOR     | The minimum ABI major version required for this guest to run.                                                  |
+/// | 16     | SMT           | 0: SMT is disallowed.<br>1: SMT is allowed.                                                               |
+/// | 17     | -             | Reserved. Must be one.                                                                                         |
+/// | 18     | MIGRATE_MA    | 0: Association with a migration agent is disallowed.<br>1: Association with a migration agent is allowed. |
+/// | 19     | DEBUG         | 0: Debugging is disallowed.<br>1: Debuggin is allowed.                                                   |
+/// | 20     | SINGLE_SOCKET | 0: Guest can be activated on multiple sockets.<br>1: Guest can only be activated on one socket.           |
+///
+#[derive(Default, Clone, Copy, Debug, Serialize, Deserialize)]
+#[repr(C)]
+pub struct SnpGuestPolicy {
+    info: u64,
+}
+
+impl SnpGuestPolicy {
+    /// The minimum ABI minor version required for this guest to run
+    pub fn abi_minor(&self) -> u8 {
+        let mask: u64 = (1 << 8) - 1;
+        (self.info & mask) as u8
+    }
+    /// The minimum ABI major version required for this guest to run
+    pub fn abi_major(&self) -> u8 {
+        let mask: u64 = ((1 << 8) - 1) << 8;
+        ((self.info & mask) >> 8) as u8
+    }
+
+    /// Bit 16 of Guest Policy's info
+    pub fn smt(&self) -> bool {
+        bit_val!(self.info, 16, u64) != 0
+    }
+
+    /// Bit 18 of Guest Policy's info
+    pub fn migrate_ma(&self) -> bool {
+        bit_val!(self.info, 18, u64) != 0
+    }
+
+    /// Bit 19 of Guest Policy's info
+    pub fn debug(&self) -> bool {
+        bit_val!(self.info, 19, u64) != 0
+    }
+
+    /// Bit 20 of Guest Policy's info
+    pub fn single_socket(&self) -> bool {
+        bit_val!(self.info, 20, u64) != 0
+    }
```

Comment:
I see there's some get methods here, would some set methods also make sense?

Reply:
I think it depends on the context, but yes, I can see setters being helpful here.

Reply:
No longer relevant. I am using `bitfield!`